cpp/src/arrow/compute/kernels/scalar_cast_numeric.cc

```diff
@@ -264,10 +264,8 @@ Status CastIntegerToFloating(KernelContext* ctx, const ExecBatch& batch, Datum*
 
 struct BooleanToNumber {
   template <typename OutValue, typename Arg0Value>
-  static OutValue Call(KernelContext*, Arg0Value val, Status*) {
-    constexpr auto kOne = static_cast<OutValue>(1);
-    constexpr auto kZero = static_cast<OutValue>(0);
-    return val ? kOne : kZero;
```

Comment:
Looks this original code is easier to read?

Reply:
Yes, that is fine. This was kind of an opinionated issue, so it is ok to ignore/remove.